content/en/docs/key-broker-service/azure/kbs-backed-by-akv.md

```diff
@@ -0,0 +1,178 @@
+
+```bash
+export SUBSCRIPTION_ID="$(az account show --query id -o tsv)"
+export RESOURCE_GROUP=my-group
+export KEYVAULT_NAME=kbs-secrets
+export CLUSTER_NAME=coco
+```
+
+## Instructions
+
+### Create Identity
+
+Create a User managed identity for KBS:
+
+```bash
+az identity create --name kbs -g "$RESOURCE_GROUP"
+export KBS_CLIENT_ID="$(az identity show -g "$RESOURCE_GROUP" --name kbs --query clientId -o tsv)"
+export KBS_TENANT_ID=$(az aks show --name "$CLUSTER_NAME" --resource-group "$RESOURCE_GROUP" --query identity.tenantId -o tsv)
+```
+
+Assign a role to access secrets:
+
+```bash
+export KEYVAULT_SCOPE=$(az keyvault show --name "$KEYVAULT_NAME" --query id -o tsv)
+az role assignment create --role "Key Vault Administrator" --assignee "$KBS_CLIENT_ID" --scope "$KEYVAULT_SCOPE"
+```
+
+### KBS identity and Service Account
+
+Workload Identity provides individual pods with IAM privileges to access Azure infrastructure resources. An azure identity is bridged to a Service Account using OIDC and Federated Credentials. Those are scoped to a Namespace, we assume we deploy the Service Account and KBS into the `default` Namespace, adjust accordingly if necessary.
+
+```bash
+export NAMESPACE=default
+export AKS_OIDC_ISSUER="$(az aks show --resource-group "$RESOURCE_GROUP" --name "$CLUSTER_NAME" --query "oidcIssuerProfile.issuerUrl" -o tsv)"
+az identity federated-credential create \
+	--name kbsfederatedidentity \
+	--identity-name kbs \
+	--resource-group "$RESOURCE_GROUP" \
+	--issuer "$AKS_OIDC_ISSUER" \
+	--subject "system:serviceaccount:${NAMESPACE}:kbs"
+```
+
+Create a Service Account object and annotate it with the identity's client id.
+
+```bash
+cat <<EOF> service-account.yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    azure.workload.identity/client-id: ${KBS_CLIENT_ID}
+  name: kbs
+  namespace: ${NAMESPACE}
+EOF
+kubectl apply -f service-account.yaml
+```
+
+## Secret Provider Class
+
+A Secret Provider Class specifies a set of secrets that should be made available to k8s workloads.
+
+```bash
+export SECRET_PROVIDER_CLASS_NAME="${KEYVAULT_NAME}-provider"
+cat <<EOF> secret-provider-class.yaml
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: $SECRET_PROVIDER_CLASS_NAME
+spec:
+  provider: azure
+  parameters:
+    usePodIdentity: "false"
+    clientID: $KBS_CLIENT_ID
+    keyvaultName: $KEYVAULT_NAME
+    objects: |
+      array:
+      - |
+        objectName: coco_one
+        objectType: secret
+      - |
+        objectName: coco_two
+        objectType: secret
+    tenantId: $KBS_TENANT_ID
+EOF
+kubectl create -f secret-provider-class.yaml
+```
+
+### Deploy KBS
+
+The Secret Provider Class can be referenced in a KBS deployment. The secrets are mounted into the storage hierarchy `default/akv`.
+
+```bash
+export KBS_IMAGE=ghcr.io/mkulke/confidential-containers/kbs:7ec54fa8f74b9bf24d84ef4a35d050d068105732-nopolicy
```

Comment:
What is the source of this image? How can users find about it?

Reply:
Ah, it makes sense to kustomize the existing kbs deployments instead.